client-js/app/scripts/app.ts

```diff
@@ -67,44 +124,35 @@ window.addEventListener('WebComponentsReady', () => {
   // Get the url for the api backend websocket connection.
   // If the apiHost variable has been set globally, use that,
   // otherwise construct a url relative to the page host.
-  var apiUrl = (window['apiHost'] || relativeWebSocketUrl()) + "/api/socket";
-
-  var socket = new rpc.JsonRpcSocket(apiUrl);
-  socket.connectionClosed.add((event) => {
-    app.connectionError = "WebSocket connection closed.";
-    app.$.errorDialog.open();
-    setTimeout(() => {
-      console.log('reloading!');
-      location.reload();
-    }, 10000);
-  });
-
-  var mgr = new manager.ManagerServiceJsonRpc(socket);
-  mgr.disconnected.add((msg) => {
-    app.connectionError = "Manager connection closed.";
-    app.$.errorDialog.open();
-    setTimeout(() => {
-      console.log('reloading!');
-      location.reload();
-    }, 10000);
-  });
-
-  var reg = new registry.RegistryServiceJsonRpc(socket);
-  var dv = new datavault.DataVaultService(socket);
-  var node = new nodeApi.NodeService(socket);
+  var apiUrl = (window['apiHost'] || relativeWebSocketUrl()) + prefix + "/api/socket";
 
   // The current activity, which encapsulates the current URL and the UI
   // rendered for that URL.
   var activity: Activity = null;
 
+  // The name of the manager to which we are currently logged in
+  var loggedInTo: string = null;
+  var services: Services = null;
+
   // Stop the current activity and start a new one. This allows activities to
   // do cleanup work before their UI is replaced, such as removing event
   // listeners.
-  async function activate(newActivity: Activity): Promise<void> {
+  async function activate(manager: string, newActivityFunc: (Services) => Activity): Promise<void> {
     try {
       if (activity) {
         await activity.stop();
       }
+      if (loggedInTo !== manager) {
```

Comment:
No, just `!==`. Think about the overall number of characters, not the number of equal signs. This will help to remember that `==` and `!=` correspond, as do `===` and `!==`.